libraries/chain/finality/finality_core.cpp

```diff
@@ -262,6 +286,7 @@ finality_core finality_core::next(const block_ref& current_block, const qc_claim
    
    assert(refs.empty() || (refs.back().block_num() + 1 == current_block.block_num())); // Satisfied by precondition 2.
    assert(refs.empty() || (refs.back().timestamp < current_block.timestamp)); // Satisfied by precondition 2.
+   assert(refs.empty() || (refs.back().finalizer_policy_generation <= current_block.finalizer_policy_generation)); // Satisfied by precondition 2.
```

Comment:
I think we could have:
```suggestion
   assert(refs.empty() || (current_block.finalizer_policy_generation - refs.back().finalizer_policy_generation <= 1)); // Satisfied by precondition 2.
```